src/facade/dragonfly_connection.cc

```diff
@@ -515,8 +515,24 @@ void Connection::ConnectionFlow(FiberSocketBase* peer) {
         ec = ec2;
       }
     }
-    error_code ec2 = peer->Shutdown(SHUT_RDWR);
+
+    // Shutdown the servers side of the socket to send a FIN to the client
+    // then keep draining the socket (discarding any received data) until
+    // the client closes the connection.
+    //
+    // Otherwise the clients write could fail (or block), so they would never
+    // read the above protocol error (see issue #1327).
+    error_code ec2 = peer->Shutdown(SHUT_WR);
     LOG_IF(WARNING, ec2) << "Could not shutdown socket " << ec2;
+    if (!ec2) {
```

Comment:
have you checked with the snippet from #1327 that EPIPE indeed disappears and the error is now visible?

Reply:
Ah, sorry missed the PR description 😆